src/main/kotlin/no/item/xp/plugin/util/GenerateInterfaceName.kt

```diff
@@ -0,0 +1,23 @@
+package no.item.xp.plugin.util
+
+import arrow.core.Either
+import java.io.File
+import java.lang.Exception
+
+@Suppress("FunctionName")
+fun GenerateInterfaceName(fileName: File): Either<Throwable, String> {
+  val filePath: String? = pathComponent(fileName.absolutePath)
+  val newFileName: String = filePath + "\\" + fileName.nameWithoutExtension + ".ts"
+  val file = File(newFileName)
+  val isNewFileCreated: Boolean = file.createNewFile()
+  return if (isNewFileCreated) {
+    Either.right(newFileName)
+  } else {
+    Either.left(Exception("File already exists: $newFileName"))
+  }
+}
+
+fun pathComponent(filename: String): String {
```

Comment:
Renamed to "getPathFromFile()"
-1 is a check because in String.kt there is this little bit concerning lastIndexOf()
"@return An index of the first occurrence of [string] or -1 if none is found."